internal/ratelimit/limiter.go

```diff
@@ -14,3 +19,63 @@ func NewLimiter(limits []*rate.Limit, maxEntries int) (*rate.Limiter, error) {
 		rate.WithQuotaStorageCapacityMetric(rateLimitQuotaStorageCapacity),
 	)
 }
+
+// limit is a representation of a rate.Limit that is used when emitting a sys
+// event to report the rate limit configuration.
+type limit struct {
+	Resource  string `json:"resource"`
+	Action    string `json:"action"`
+	Per       string `json:"per"`
+	Unlimited bool   `json:"unlimited"`
+	Limit     uint64 `json:"limit"`
+	Period    string `json:"period"`
+}
+
+type (
+	actionLimits    []limit
+	resourceActions map[string]actionLimits
+	resources       map[string]resourceActions
+)
+
+// WriteLimitsSysEvent writes a sys event that contains all of the provided
+// rate limits.
+func WriteLimitsSysEvent(ctx context.Context, limits []*rate.Limit, maxEntries int) error {
```

Comment:
When can this function return an error?

Reply:
Hmmm. I don't think it can. I'll update it.